lib/mayaUsd/ufe/UsdUndoRenameCommand.cpp

```diff
@@ -131,8 +133,13 @@ bool UsdUndoRenameCommand::renameRedo()
         _newName = uniqueName(childrenNames, _newName);
     }
 
+    // names are not allowed to start to digit numbers
+    if(std::isdigit(_newName.at(0))){
+        _newName = prim.GetName();
+    }
+
     // all special characters are replaced with `_`
-    const std::string specialChars{"~!@#$%^&*()-=+,.?`':{}|<>[]/"};
+    const std::string specialChars{"~!@#$%^&*()-=+,.?`':{}|<>[]/' '"};
```

Comment:
Space also needs to be replaced with under score.

Reply:
Think you added the space AND the two single quotes, which you don't need because you're in a string.  In fact, the single quote is already in your string, so now you have it 3 times...

Reply:
@ppt-adsk Duh!! Thanks.  I could use std::isspace for sake of readability but that's something for later.  All these string manipulation and logics could also go to somewhere like stringUtils.h later on.

See bb59f68ab5fa7d55a44d264dce73a67a2384ce89